fs/rc/rc.go

```diff
@@ -130,6 +117,81 @@ func (s *server) handler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	fs.Debugf(nil, "form = %+v", r.Form)
+
+	w.Header().Add("Access-Control-Allow-Origin", "*")
+	//echo back headers client needs
+	reqAccessHeaders := r.Header.Get("Access-Control-Request-Headers")
+	w.Header().Add("Access-Control-Allow-Headers", reqAccessHeaders)
+
+	if r.Method == "POST" {
+		s.handlePost(w, r, path, in)
+	} else if r.Method == "OPTIONS" {
+		s.handleOptions(w, r, path, in)
+	} else {
+		writeError(errors.Errorf("method %q not allowed - POST or OPTIONS required", r.Method), http.StatusMethodNotAllowed)
+		return
+	}
+}
+
+func (s *server) handlePost(w http.ResponseWriter, r *http.Request, path string, in Params) {
+	//TODO: find elegant way to extract this from handlers
+	writeError := func(err error, status int) {
+		fs.Errorf(nil, "rc: %q: error: %v", path, err)
+		w.WriteHeader(status)
+		err = WriteJSON(w, Params{
+			"error": err.Error(),
+			"input": in,
+		})
+		if err != nil {
+			// can't return the error at this point
+			fs.Errorf(nil, "rc: failed to write JSON output: %v", err)
+		}
+	}
+
+	// Find the call
+	call := registry.get(path)
+	if call == nil {
+		writeError(errors.Errorf("couldn't find method %q", path), http.StatusMethodNotAllowed)
+		return
+	}
+
+	fs.Debugf(nil, "rc: %q: with parameters %+v", path, in)
+	out, err := call.Fn(in)
+	if err != nil {
+		writeError(errors.Wrap(err, "remote control command failed"), http.StatusInternalServerError)
+		return
+	}
+
+	fs.Debugf(nil, "rc: %q: reply %+v: %v", path, out, err)
+	err = WriteJSON(w, out)
+	if err != nil {
+		// can't return the error at this point
+		fs.Errorf(nil, "rc: failed to write JSON output: %v", err)
+	}
+}
+func (s *server) handleOptions(w http.ResponseWriter, r *http.Request, path string, in Params) {
+	//TODO: find elegant way to extract this from handlers
+	writeError := func(err error, status int) {
+		fs.Errorf(nil, "rc: %q: error: %v", path, err)
+		w.WriteHeader(status)
+		err = WriteJSON(w, Params{
+			"error": err.Error(),
+			"input": in,
+		})
+		if err != nil {
+			// can't return the error at this point
+			fs.Errorf(nil, "rc: failed to write JSON output: %v", err)
+		}
+	}
+
+	// For options request we do nothing, just noop and reflect client input
+	call := registry.get("rc/noop")
```

Comment:
My reading of what OPTIONS is supposed to do indicates it should return an empty body.  So I think just returning rather than running `rc/noop` will do, which also means no possibility of returning an error so no error handling needed!